config.go

```diff
@@ -479,6 +479,10 @@ type Config struct {
 	// 2. Use the table's primary key column as the pagination column. Fail if the primary key is not numeric or is a composite key without a FallbackColumn specified.
 	// 3. Use the FallbackColumn pagination column, if configured. Fail if we cannot find this column in the table.
 	CascadingPaginationColumnConfig *CascadingPaginationColumnConfig
+
+	// SQL query comments to differentiate Ghostferry's binlog events
+	// Optional: defaults to empty string (no comments)
+	QueryComment string
```

Comment:
By moving this piece of configuration to within the `DatabaseConfig`, we can remove the need to pass it around to the various constructors that will use it and just reference it internally in the `SqlDB` method and in `ghostferry_db.go`.

I would also recommend renaming this to `Marginalia` as it is consistent with our other naming internally. If we think it's a problem that this is confusing because it's not using ActiveRecord (https://github.com/shopify/marginalia), then we can use a name like `Annotation`.